File: 3804-number-of-centered-subarrays/3804-number-of-centered-subarrays.java
class Solution {
    public int centeredSubarrays(int[] nums) {
        int ans=0;
        for(int i=0;i<nums.length;i++){
            int s=0;
            HashSet<Integer> set=new HashSet<>();
            for(int j=i;j<nums.length;j++){
                set.add(nums[j]);
                s+=nums[j];
                if(set.contains(s)) ans++;
            }
        }
        return ans;
    }
}

File: 3804-number-of-centered-subarrays/3804-number-of-centered-subarrays.cpp
class Solution {
public:
    int centeredSubarrays(vector<int>& nums) {
        int ans=0;
        unordered_set<int> set;
        for(int i=0;i<nums.size();i++){
            int s=0;
            set.clear();           
            for(int j=i;j<nums.size();j++){
                set.insert(nums[j]);
                s+=nums[j];
                if(set.find(s)!=set.end()) ans++;
            }
        }
        return ans;
    }
};